bubble_sort.rb

```diff
@@ -0,0 +1,40 @@
+def bubble_sort(numbers)
+  if numbers.is_a?(Array) && numbers.any?
+    swap = true
+    while swap == true
+      swap = false
+      (numbers.length - 1).times do |i|
+        if numbers[i] > numbers[i + 1]
+          numbers[i], numbers[i + 1] = numbers[i + 1], numbers[i]
+          swap = true
+        end
+      end
+    end
+    print numbers
+  else
+    puts 'No numbers given.'
+  end
+end
+
+def bubble_sort_by(array)
+  swap = true
+  if array.is_a?(Array) && array.any?
+    while swap == true
+      swap = false
+      (array.size - 1).times do |i|
+        comparator = yield array[i], array[i + 1]
+        if comparator
+          array[i], array[i + 1] = array[i + 1], array[i]
+          swap = true
+        end
```

Comment:
You should check if the comparator is greater than zero, before proceeding to swap.
What you have currently will proceed to swap every time except when comparator value is false or nil.